rig/machine_control/bmp_controller.py

```diff
@@ -168,9 +168,8 @@ def _send_scp(self, cabinet, frame, board, *args, **kwargs):
 
         return connection.send_scp(length, 0, 0, board, *args, **kwargs)
 
-    @ContextMixin.use_contextual_arguments
-    def get_software_version(self, cabinet=Required, frame=Required,
-                             board=Required):
+    @ContextMixin.use_contextual_arguments()
```

Comment:
It doesn't _look_ magic from the coder point of view though, I don't think.  Presumably:

```python
def use_contextual_arguments(f=None, **kwarg_only_args_defaults):
    def decorator(f):
        # ...
    if f is not None:
        return decorator(f)
    else:
        return decorator
```

Would do it?
```